judge/forms.py

```diff
@@ -69,13 +69,19 @@ def __init__(self, *args, **kwargs):
 
 class ProblemSubmitForm(ModelForm):
     source = CharField(max_length=65536, widget=AceWidget(theme='twilight', no_ace_media=True))
+    judge = ChoiceField(choices=(), widget=forms.HiddenInput(), required=False)
 
     def __init__(self, *args, **kwargs):
+        can_edit_problem = kwargs.pop('can_edit_problem', None)
         super(ProblemSubmitForm, self).__init__(*args, **kwargs)
         self.fields['language'].empty_label = None
         self.fields['language'].label_from_instance = attrgetter('display_name')
         self.fields['language'].queryset = Language.objects.filter(judges__online=True).distinct()
 
+        if can_edit_problem:
+            self.fields['judge'].widget = Select2Widget(attrs={'style': 'width:150px', 'data-placeholder': _('any')})
+            self.fields['judge'].choices = tuple(Judge.objects.filter(online=True).values_list('name', 'name'))
```

Comment:
Does this filter for judges the problem is available on?

Reply:
This has been fixed.